_posts/pytorch_hub.md

```diff
@@ -0,0 +1,163 @@
+
+```python
+dependencies = ['torch', 'tqdm', 'boto3', 'requests', 'regex']
+
+from hubconfs.bert_hubconf import (
+    bertTokenizer,
+    bertModel,
+    bertForNextSentencePrediction,
+    bertForPreTraining,
+    bertForMaskedLM,
+    bertForSequenceClassification,
+    bertForMultipleChoice,
+    bertForQuestionAnswering,
+    bertForTokenClassification
+)
+from hubconfs.gpt_hubconf import (
+    openAIGPTTokenizer,
+    openAIGPTModel,
+    openAIGPTLMHeadModel,
+    openAIGPTDoubleHeadsModel
+)
+```
+
+Each model then requires an entrypoint to be created. Here is a code snippet to specify an entrypoint of the ```bertForMaskedLM``` model, which returns the pre-trained model weights.
+
+```python
+def bertForMaskedLM(*args, **kwargs):
+    """
+    BertForMaskedLM includes the BertModel Transformer followed by the
+    pre-trained masked language modeling head.
+    Example:
+      ...
+    """
+    model = BertForMaskedLM.from_pretrained(*args, **kwargs)
+    return model
+```
+
+With these in place and a pull request based on the template [here](https://github.com/pytorch/hub/blob/master/docs/template.md) you are all set to go. We may work with you to refine your code snippet but this is only because it is used as part of continuous integration (CI) and will help to identify any breaking changes in a timely manner. Your model will soon appear on [Pytorch hub webpage](https://pytorch.org/hub) for all users to explore.
+
+
+## [User] Workflow
+
+As a user, PyTorch Hub allows you to follow a few simple steps and do things like: 1) explore available models; 2) load a model; and 3) understand what methods are available for any given model. Let's walk through some examples of each.
+
+### Explore available entrypoints.
+
+Users can list all available entrypoints in a repo using the ```torch.hub.list()``` API. In this case, we are exploring the HuggingFace Bert models and we can see that there are several models available for us to try out.
+
+```python
+>>> torch.hub.list('huggingface/pytorch-pretrained-BERT')
+>>>
+['bertForMaskedLM',
+ 'bertForNextSentencePrediction',
+ ...
+ 'openAIGPTModel',
+ 'openAIGPTTokenizer']
+ ```
+
+Note that PyTorch Hub also allows auxillary entrypoints (other than pretrained models), e.g. ```bertTokenizer``` for preprocessing, to make the user workflow smoother.
+
+
+### Load a model
+
+Now that we know which models are available in the Hub, users can load a model entrypoint using the ```torch.hub.load()``` API. Because dependencies are taken care of by PyTorch Hub, this only requires a single command without the need to install a wheel.
+
+```python
```

Comment:
In fact I realize we should add sth for `torch.hub.help()` here, like: 
`hub.help()` shows how to **instantiate** a model.
And add this line in the code block below.

```
print(torch.hub.help('huggingface/pytorch-pretrained-BERT', 'bertForMaskedLM'))
```